internal/controller/pdb_to_pdbwatcher_controller.go

```diff
@@ -0,0 +1,133 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	types "github.com/paulgmiller/k8s-pdb-autoscaler/api/v1"
+	policyv1 "k8s.io/api/policy/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// PDBToPDBWatcherReconciler reconciles a PodDisruptionBudget object.
+type PDBToPDBWatcherReconciler struct {
+	client.Client
+	Scheme   *runtime.Scheme
+	Recorder record.EventRecorder
+}
+
+// Reconcile reads the state of the cluster for a PDB and creates/deletes PDBWatchers accordingly.
+func (r *PDBToPDBWatcherReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	// Fetch the PodDisruptionBudget object based on the reconcile request
+	var pdb policyv1.PodDisruptionBudget
+	err := r.Get(ctx, req.NamespacedName, &pdb)
+	if err != nil {
+		// If the PDB is deleted, we should delete the corresponding PDBWatcher.
+		// First, check if the PDBWatcher exists
+		var pdbWatcher types.PDBWatcher
+		err := r.Get(ctx, req.NamespacedName, &pdbWatcher)
+		if err == nil {
+			// Delete PDBWatcher if PDB is deleted
```

Comment:
a pdbwatcher already  gets marked as degraded if there is no corresponding pdb. So you could leave it _unless_ this controller is what created it. (need some sort of owner/createdby concept here)

Reply:
and as with comment in deployment_to_pdb_controller you can miss deletes here. Thats okay just be aware.

Reply:
Realized pdbwatcher itself should try and scale back to origional replicas when its deled.